### ISSUE TITLE
🐛 [Bug Report] Ammo for Rumbleshot quest unable to turn in.

### ISSUE THREAD

**Issue by MilkmanMHOC** [cmangos/issues#3200]
### Bug Details

Taking the quest works, go get his ammo works, but after finding Hegnar Rumbleshot there is no ? nor can you talk to him to turn in the quest he simple acts as a vendor when talked too.

https://www.wowhead.com/classic/quest=5541/ammo-for-rumbleshot

Has the details of the quest.

### Steps to Reproduce

1.
2.
3.
4.

### Core SHA1 Commit Hash

9098c0b44ff496e476b9f6bec6b551764b29a151

### Database SHA1 Commit Hash

3bf58535494a0ee3c466f935f0bd711acd5472ec

### Operating System

Windows 7 (yup it works)

### Client Version

1.12.1 (Classic)

**Comment by AnonXS** [comment]
During the RP Event you cant turn the quest in as the flag is removed. (.q add 5541 .q com 5541)

![grafik](https://user-images.githubusercontent.com/19734826/200148377-6db1b2b5-0cd6-477b-adfa-7a5e0e3c5893.png)

if you wait ~5mins until they return to the fire you can turn the quest in:

Rumbleshot doesnt seem to do anything important so i'm not sure why we remove the questgiver flag when this rp script starts.  (.go c id 1243)

![grafik](https://user-images.githubusercontent.com/19734826/200148425-3f3e40c6-ba64-4a17-8e46-f6d1be612139.png)
![grafik](https://user-images.githubusercontent.com/19734826/200148431-ba76a6a6-5b28-44eb-8cdc-4473b812f749.png)
![grafik](https://user-images.githubusercontent.com/19734826/200148440-fc5284bf-4235-4388-a25d-0a80410487b7.png)

maybe this could be bypassed now by setting interactionpausetimer to 0.
event also repeats very frequently so its likely to run into this "issue".

**Comment by MilkmanMHOC** [comment]
I have no proof of what it was like originally.. so not sure if should be changed?  And yes its completable as you said, just need to wait.

**Comment by Grz3s** [comment]
Hegnar Rumbleshot - starts RP event... (this is how i made it some years ago)
Problem occured when someone did show proof ...that same event happens as afterscript for this quest. So to prevent buggin everything i removed his NPCFLAG until event is completed. (i dont remember if this is how it suppossed to be - sor).
With current techs we got now i could prob do it in diff way.
If someone has active blizz acc (classic to WotLK only) and spare 15 mins:
- go there and check if this event really starts on OOC timer... or only as afterscript for q. 

BTW:
found commit where i added that NPCflag check ... and connected it to quest:
https://github.com/cmangos/wotlk-db/commit/e897dceaed43f540ab49726824774af5e1d5326e
original script was added in UDB 407
https://github.com/unified-db/Database/releases/tag/v0.12.2-407
